components/movementsensor/wheeledodometry/wheeled_odometry.go

```diff
@@ -134,11 +129,10 @@ func NewWheeledOdometry(
 		if err != nil {
 			return nil, err
 		}
-		ok := properties.PositionReporting
-		if !ok {
+		if !properties.PositionReporting {
 			return nil, motor.NewPropertyUnsupportedError(properties, newConf.LeftMotors[i])
 		}
-		o.leftMotors = append(o.leftMotors, motorDep)
+		o.motors[i].left = motorDep
```

Comment:
maybe we can end up with
```suggestion
		o.motors = motorPair{left, right}
```
someway

Reply:
yeah that would be ideal let me try and rework it